Types/Module.cs

```diff
@@ -403,7 +403,20 @@ public static CkanModule FromJson(string json)
             }
             // All good! Return module
             return newModule;
-        }            
+        }
+        public override bool Equals(object obj)
+        {
+            var other = obj as Module;
+            return other != null
+            ? identifier.Equals(other.identifier) && version.Equals(other.version)
+            : base.Equals(obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return identifier.GetHashCode();
```

Comment:
Aaah! Looks [at the docs](https://msdn.microsoft.com/en-us/library/system.object.gethashcode%28v=vs.110%29.aspx) equal hash-codes indicate they *may* be equal. Unequal hash codes indicate they're definitely not. So this is a promise that mods will never be equal unless their identifiers match at least.

I may modify this slightly if we can efficiently xor this with the `Version` hashcode, but it's not a blocker if I can't do that.

- [ ] (Optional) xor identifier and version for hashcode.